decoder/stats.go

```diff
@@ -375,6 +422,158 @@ func updatePokemonStats(old *Pokemon, new *Pokemon, areas []geo.AreaName) {
 	}
 }
 
+func updateRaidStats(old *Gym, new *Gym, areas []geo.AreaName) {
+	if len(areas) == 0 {
+		areas = []geo.AreaName{
+			{
+				Parent: "world",
+				Name:   "world",
+			},
+		}
+	}
+
+	locked := false
+
+	// Loop though all areas
+	for i := 0; i < len(areas); i++ {
+		area := areas[i]
+
+		if new.RaidSpawnTimestamp.ValueOrZero() > 0 &&
+			(old == nil || old.RaidLevel != new.RaidLevel ||
+				old.RaidPokemonId != new.RaidPokemonId ||
+				old.RaidSpawnTimestamp != new.RaidSpawnTimestamp) {
```

Comment:
I think we don't need the check for raid level and pokemon id... it can happen that the same legendary raid is spawning two times e.g.. we could simply check for a changed raid spawn timestamp or raid end timestamp :)

Reply:
Hopefully this should be good now in case of Back-to-back raids.